File: frontend/app/src/main/java/com/example/geoquiz_frontend/data/remote/dtos/auth/RegisterRequest.java
package com.example.geoquiz_frontend.data.remote.dtos.auth;

import com.example.geoquiz_frontend.domain.entities.UserStats;
import com.google.gson.annotations.SerializedName;

public class RegisterRequest {
    private String userName;
    private String email;
    private String password;
    @SerializedName("stats")
    private UserStatsRequest stats;

    public RegisterRequest(String userName, String email, String password) {
        this.userName = userName;
        this.email = email;
        this.password = password;
        this.stats = null;
    }

    public RegisterRequest(String userName, String email, String password, UserStats userStats) {
        this.userName = userName;
        this.email = email;
        this.password = password;
        if (userStats != null) {
            this.stats = new UserStatsRequest(userStats);
        }
    }
}

File: backend/GeoQuiz_backend/GeoQuiz_backend/Application/DTOs/Auth/RegisterRequest.cs
using GeoQuiz_backend.Domain.Entities;

namespace GeoQuiz_backend.Application.DTOs.Auth
{
    public class RegisterRequest
    {
        public string UserName { get; set; } = null!;
        public string Email { get; set; } = null!;
        public string Password { get; set; } = null!;
        public UserStatsDto? Stats { get; set; } 
    }
}
